floss/language_identifier.py

```diff
@@ -0,0 +1,68 @@
+# Copyright (C) 2023 Mandiant, Inc. All Rights Reserved.
+import pefile
+
+import floss.logging_
+from floss.logging_ import DebugLevel
+
+logger = floss.logging_.getLogger("floss")
+
+
+def is_go_bin(sample: str) -> bool:
+    """
+    Check if the binary given is compiled with GO compiler or not
+    it checks the magic header of the pclntab structure -pcHeader-
+    the magic values varies through the version
+    reference:
+    https://github.com/0xjiayu/go_parser/blob/865359c297257e00165beb1683ef6a679edc2c7f/pclntbl.py#L46
+    """
+    try:
+        pe = pefile.PE(sample)
+    except:
+        logger.debug("NOT valid PE header")
+        return False
+
+    go_magic = [
+        b"\xf0\xff\xff\xff\x00\x00",
+        b"\xfb\xff\xff\xff\x00\x00",
+        b"\xfa\xff\xff\xff\x00\x00",
+        b"\xf1\xff\xff\xff\x00\x00",
+    ]
+
+    # look for the .rdata section first
+    for section in pe.sections:
+        if ".rdata" in section.Name.decode("utf-8").rstrip("\x00"):
+            section_va = section.VirtualAddress
+            section_size = section.SizeOfRawData
+            for magic in go_magic:
+                if magic in section.get_data(section_va, section_size):
+                    pclntab_va = section.get_data(section_va, section_size).index(magic) + section_va
+                    if verify_pclntab(section, pclntab_va):
+                        return True
```

Comment:
it might be expensive to fetch the section data (in practice maybe not that bad, but still), so lets pull the computation out of the inner loop

```suggestion
            section_data = section.get_data(section_va, section_size)
            for magic in go_magic:
                if magic in section_data:
                    pclntab_va = section_data.index(magic) + section_va
                    if verify_pclntab(section, pclntab_va):
                        return True
```